provider/pkg/resources/cfn_custom_resource.go

```diff
@@ -0,0 +1,681 @@
+package resources
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/aws/aws-lambda-go/cfn"
+	"github.com/golang/glog"
+	"github.com/google/uuid"
+	"github.com/pulumi/pulumi-aws-native/provider/pkg/client"
+	"github.com/pulumi/pulumi-aws-native/provider/pkg/naming"
+	"github.com/pulumi/pulumi-go-provider/resourcex"
+	pschema "github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/urn"
+)
+
+// This is the default timeout for custom resource operations in CloudFormation
+const DefaultCustomResourceTimeout = 60 * time.Minute
+
+var lambdaFunctionArnRegex = regexp.MustCompile(`^arn:[^:]+:lambda:[^:]+:[^:]+:function:[a-zA-Z0-9-_]+(:[a-zA-Z0-9-_]+)?$`)
+
+type Clock interface {
+	Now() time.Time
+	Since(time.Time) time.Duration
```

Comment:
Nit: Go likes minimal interfaces so you only really need Now(), as Since can be computed from Now in a `func`.

Reply:
Yes, but that would clutter the implementation. The purpose of this interface is to make the code testable. I'd rather just pass through to the stdlib than add custom code

Reply:
Mhm, I guess this is a bit of a moot point from my side given that you can subtract times and get the duration that way

Reply:
Yeah it's really a small nit, feel free to disregard.